CMSIS/CoreValidation/Source/CV_CoreFunc.c

```diff
@@ -307,6 +313,9 @@ void TC_CoreFunc_IPSR (void) {
 #if defined(__CC_ARM)
 #define SUBS(Rd, Rm, Rn) __ASM volatile("SUBS " # Rd ", " # Rm ", " # Rn)
 #define ADDS(Rd, Rm, Rn) __ASM volatile("ADDS " # Rd ", " # Rm ", " # Rn)
+#elif defined(__ti__)
```

Comment:
Why is this duplication of the else-tree required?
Is it because `__GNUC__` is defined?

Reply:
Perhaps we need to move these defines into the compiler header?

Reply:
Yes, __GNUC__ is also defined.  However, I can just add a !defined(__ti__) to the GNUC clause and get the same behavior without duplicating the macros.  it's worth exploring moving these defines into the compiler headers.